### PR TITLE
Add code coverage workflow.

### PR DESCRIPTION
**Pull request by hoolioh**
# What does this PR do?
This PR adds a new workflow file to generate code coverage reports and upload the results to codecov. Once the first report is uploaded each PR will have a comment showing the diff with the previous state. The diff will present changes in the following components:

- crashtracker
- data-pipeline
- data-pipeline-ffi
- ddcommon
- ddcommon-ffi
- ddtelemetry
- ipc
- profiling
- profiling-ffi
- serverless
- sidecar
- sidecar-ffi
- spawn_worker
- trace-mini-agent
- trace-normalization
- trace-obfuscation
- trace-protobuf
- trace-utils

### CODE REVIEW

**Review by ekump — APPROVED** [review]

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+comment:
+  layout: "condensed_header, diff, flags, components"
+  
+component_management:
+  individual_components:
+    - component_id: crahstracker  # this is an identifier that should not be changed
+      name: crahstracker  # this is a display name, and can be changed freely
+      paths:
+        - crahstracker
```
Comment (danielsn) [inline .codecov.yml:9]:
```suggestion
    - component_id: crashtracker  # this is an identifier that should not be changed
      name: crashtracker  # this is a display name, and can be changed freely
      paths:
        - crashtracker
```

**Review by danielsn — COMMENTED** [review]
Thanks for doing this.  Please fix the typo before merging!

**Review by bantonsson — APPROVED** [review]

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  coverage:
+    runs-on: ubuntu-latest
+    env:
+      CARGO_TERM_COLOR: always
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust
+        run: rustup update stable
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
```
Comment (bantonsson) [inline .github/workflows/coverage.yml:19]:
```suggestion
        uses: taiki-e/install-action@cargo-llvm-cov@v0.6.9
```

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Generate code coverage
+        run: cargo llvm-cov --all-features --workspace --lcov --output-path lcov.info
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
+          slug: hoolioh/libdatadog
```
Comment (bantonsson) [inline .github/workflows/coverage.yml:26]: I think the `slug` setting needs to be removed, and the comment for `token` is not true.

**Review by bantonsson — COMMENTED** [review]
Some minor things that I noticed when enabling the workflows.

**Pull request merged**